### ISSUE TITLE
[S3854] Raise wrong "must be invoked with 2 arguments." error on destructuring

### ISSUE THREAD

**Issue by AMorgaut** [SonarSource/SonarJS#886]
**I want to report a bug.**

**SonarJS version:**  3.3.0.5702<br>
**SonarQube version:** 5.6.7<br>

**Rule key:** S3854

**Reproducer** 
```javascript
const defaultOptions = {};
class Foo {
    constructor(name, options = {}) {
        this.name = name;
        this.options = Object.assign({}, defaultOptions, options);
    }
}
class Bar extends Foo {
    constructor(...args) {
        super(...args);
    }
}
function getBar(name, options = {}) {
    return new Bar(name, options);
}
```

**Logs**
```
super() must be invoked with 2 arguments.
Rule: javascript:S3854
```

**Expected behavior**

There should be no error as super will correctly receive the right number of arguments

**Comment by vilchik-elena** [comment]
[RSPEC-3854](https://jira.sonarsource.com/browse/RSPEC-3854)

**Comment by saberduck** [comment]
There is also an issue when `super` has parameters with default values. 
```
class Foo {
    constructor(name, options = {}) {
        this.name = name;
        this.options = Object.assign({}, defaultOptions, options);
    }
}

class Baz extends Foo {
  constructor() {
    super("blah"); // issue raised here
  }
}
```
